libcxxabi/src/demangle/ItaniumDemangle.h

```diff
@@ -5741,14 +5741,12 @@ struct FloatData<double>
 template <>
 struct FloatData<long double>
 {
-#if defined(__mips__) && defined(__mips_n64) || defined(__aarch64__) || \
-    defined(__wasm__) || defined(__riscv) || defined(__loongarch__) || \
-    defined(__ve__)
-    static const size_t mangled_size = 32;
-#elif defined(__arm__) || defined(__mips__) || defined(__hexagon__)
-    static const size_t mangled_size = 16;
-#else
-    static const size_t mangled_size = 20;  // May need to be adjusted to 16 or 24 on other platforms
+#if __LDBL_MANT_DIG__ == 113
+  static const size_t mangled_size = 32;
+#elif __LDBL_MANT_DIG__ == 53
+  static const size_t mangled_size = 16;
+#else // __LDBL_MANT_DIG__ == 64
```

Comment:
I think it'd be better to check the mantissa here as well and `#error` if there is a new size. Otherwise we might be silently wrong on some platforms.

Reply:
That sounds like a good idea, will fix.

However - I realized that while we don't support building libcxxabi with compilers that aren't GCC compatible, we also do copy this file into `llvm/include/llvm/Demangle/ItaniumDemangle.h`, which is built as part of the host build of LLVM - and that can be built with MSVC as well. (And MSVC doesn't define any of the arch defines that we defined before either.)

MSVC only has `long double == double` for all currently known architectures (and clang-cl matches this), so I can add an ` || defined(_MSC_VER)` for that.